specs/devexperience/DevExperience.md

```diff
@@ -0,0 +1,181 @@
+10.	The existing Publish UI will be enabled for creating and publishing app packages to the Microsoft Store.
+
+
+## 5.0 - Developer Workflow
+This section goes into more fine grain detail about the developer experience that will come with developers who are using Project Reunion. It will expand off the Key Design Elements and Principles sections, and articulate where each of those comes into play in the overall story of Reunion.
+
+### 5.1 – Migration and Compat
+This section will highlight the basic workflow for developers looking to use Reunion, as well as highlight some basic high-level requirements that the tooling must guarantee. The section on MSIX packaging will go into more detail on how we will simplify the developer story, while maintaining compat at the same time.
+
+#### 5.1.1 - Migrating Existing Win32 Apps
+App developers which are migrating existing Win32 Apps will do the following:
+1.	Install Reunion VSIX
+2.	Add package reference to Reunion
+At this point, the developer can stop and have an unpackaged app, with the ability to use MRT. None of the MSIX package tooling will run, because there is no existence of a Package.appxmanifest file. If the developer would like to create a packaged Desktop app, then they do the following:
+3.	Add Package.appxmanifest file to the project
+After this step, if they’d prefer to create a Sparse Package, they would add this to their project file:
+<WindowsPackageType>Sparse</WindowsPackageType>
+Question: could we update try convert to give existing Win32 apps Sparse by default?
+
+### 5.1.2 - Migrating Existing UWP Apps
+Existing UWP applications will require being converted to WinUI3 applications first. The following are the steps a developer will run in order to migrate their apps:
+1.	Run the try-convert tool
+a.	If the user wants a desktop app, they will then update <WindowsPackageType> in their project file.
+2.	Fix up any compilation errors that may exist
+
+### 5.1.3 - Other Compat/Migration Concerns
+None of the tooling will be WinUI/Reunion specific, and the tooling could be used for all existing UWP apps. However, the hypothesis is that trying to move existing UWP apps to the new tooling pipeline would result in noise and over-complicate the story, as well as require more documentation updates. Therefore, we will not be advertising it as such. However, one of the key components of the tooling will be that it is easily migratable, and therefore we cannot require users to make any changes to their existing build customizations. By design, this forces the tooling to maintain compatibility.
+
+With that being said, one thing that will be supported is users having a mix of old UWP and newer Reunion projects in the same .sln file or repository. This will allow developers to migrate slowly at their own pace. In order to do this, the task assemblies that ship as part of Reunion must be renamed. Otherwise, MSBuild will mistakenly use the wrong assemblies.
```

Comment:
>...the task assemblies that ship as part of Reunion must be renamed.

Could you elaborate? What is a 'task assembly'? Do these exist today in Reunion?

Might be my VSfu is weak :-) but I'm not following this issue

Reply:
Task assemblies are the .NET assemblies that contain VS build task code.